### PR TITLE
Jetpack Manage: Add breadcrumb in the new 'Add Payment method' page.

### PR DESCRIPTION
**Pull request by jkguidaven**
The goal of this PR is to add a breadcrumb to the new 'Add Payment method' page.

Screen Shot 2024-01-10 at 9 10 41 PM

Closes https://github.com/Automattic/jetpack-genesis/issues/178

## Proposed Changes

* Add official sub-component **LayoutBreadcrumb** which wraps the existing Calypso breadcrumb. This will be usable for other use cases, so it makes sense we make this officially part of the layout. 
* Add breadcrumb in the 'Add Payment method' page.

## Testing Instructions
Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb.

* Use the Jetpack cloud live link below and go to the new Add Payment Method page (`/partner-portal/payment-methods/add?flags=jetpack/card-addition-improvements`)
* Confirm that you see the breadcrumb above the header.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by oikeme — APPROVED** [review]
Changes work as expected and the breadcrumb for "add new card" is now shown.

**Review by yashwin — COMMENTED** [review]
Thanks for working on this, and expanding the layout header to have breadcrumbs.

One thing I noticed here: The `Payment Methods` in the breadcrumbs should be a link that takes back the users to the `Payment Methods` page.

Screenshot 2024-01-11 at 8 52 59 AM

We have a similar design on the [plugin details page](https://cloud.jetpack.com/plugins/akismet):

Screenshot 2024-01-11 at 8 52 01 AM

**Comment by jkguidaven** [comment]
> One thing I noticed here: The Payment Methods in the breadcrumbs should be a link that takes back the users to the Payment Methods page.

Good point this has been addressed. 👍

**Comment by jkguidaven** [comment]
@oikeme @yashwin, thanks for the review. I made some adjustments. Can you have another look?

File: client/jetpack-cloud/sections/partner-portal/primary/payment-method-add-v2/index.tsx [hunk 1]
```diff
@@ -16,6 +17,12 @@ export default function PaymentMethodListV2() {
 		<Layout className="payment-method-add" title={ title } wide>
 			<LayoutTop>
 				<LayoutHeader>
+					<Breadcrumb
+						items={ [
+							{ label: translate( 'Payment Methods' ), href: '/partner-portal/payment-method' },
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/primary/payment-method-add-v2/index.tsx]: There is one small mistake here

```suggestion
							{ label: translate( 'Payment Methods' ), href: '/partner-portal/payment-methods' },
```

Comment (jkguidaven) [reply client/jetpack-cloud/sections/partner-portal/primary/payment-method-add-v2/index.tsx]: good catch. This has been fixed 👍

**Review by yashwin — APPROVED** [review]
LGTM now 🚢

**Pull request merged**